app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.regex_valid_email
+  before_save :downcase_email
+  validates :name, presence: true,
+                   length: { maximum: Settings.users.user_name.maximum }
+  validates :email, presence: true,
+                    length: { Settings.users.mail.maximum },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: true
+
+  private
+
+  def downcase_email
+    self.email.downcase!
```

Comment:
```suggestion
    email.downcase!
```
.downcase! khác gì .downcase nhỉ